dotnet/src/Skills/Skills.OpenAPI/Builders/OperationComponentBuilderFactory.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.SemanticKernel.Skills.OpenAPI.Builders.Query;
+
+namespace Microsoft.SemanticKernel.Skills.OpenAPI.Builders;
+
+/// <summary>
+/// Defines factory for creating builders used to construct various components(path, query string, payload, etc...) of REST API operations.
+/// </summary>
+internal class OperationComponentBuilderFactory : IOperationComponentBuilderFactory
```

Comment:
I feel like this could be overkill... URLs are pretty standard and .net already has good tools. Do we expect anyone will want to override?

Reply:
It's a little bit broader than URL building. The plan is to have a builder for each REST API operation component:
- Server URL: It can be parameterized, and the arguments for the parameters need to be resolved.
- Resource path: It can have various serialization styles, and the arguments for the parameters need to be resolved.
- Query string: It can have various serialization styles, and the arguments for the parameters need to be resolved.
- Headers: They can be serialized differently, and the arguments for the parameters need to be resolved.
- Body: It can have various media types such as application/json, text/plain, form, etc. The body needs to be built dynamically in a way specific to the media type, and the arguments for the parameters need to be resolved.
- Cookie: It can be serialized differently, and the arguments for the parameters need to be resolved."

On top of that, it's likely that the serialization style/format may differ between OpenAPI document versions. Therefore, it is better have a good abstraction/structure that enables us to handle serialization-version permutations with ease, rather than accumulating all the code within the `RestApiOperation` or `RestApiOperationRunner` classes, which would clutter them with `if(param.Type == "Query" && param.Style == "Form" && doc.Version == "2.0"...)` code and make them unmaintainable over time.

More details about various OpenAPI pa
* https://swagger.io/docs/specification/describing-parameters/
* https://swagger.io/docs/specification/serialization/
* https://swagger.io/docs/specification/describing-request-body/